inbc-program/inbc/parser/source_app_parser.py

```diff
@@ -19,15 +27,16 @@ def application_add(args: argparse.Namespace) -> str:
     manifest = ('<?xml version="1.0" encoding="utf-8"?>' +
                 '<manifest>' +
                 '<type>source</type>' +
-                '<applicationSource>' +
-                '<add><gpg>'
-                '{0}' +
-                '{1}'
-                '</gpg><repo><repos>').format(create_xml_tag(arguments, "uri"),
-                                              create_xml_tag(arguments, "keyname"))
+                '<applicationSource>' + '<add>')
```

Comment:
**Enhancement**
**Suggestion:** Use XML libraries to build XML documents instead of string concatenation to prevent potential issues with special characters and improve readability.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    from xml.etree.ElementTree import Element, SubElement, tostring
    from xml.dom import minidom

    manifest = Element('manifest')
    manifest.set('version', '1.0')
    manifest.set('encoding', 'utf-8')
    source_type = SubElement(manifest, 'type')
    source_type.text = 'source'
    app_source = SubElement(manifest, 'applicationSource')
    add_element = SubElement(app_source, 'add')
    # Further XML construction should continue using the ElementTree API
    manifest_str = minidom.parseString(tostring(manifest)).toprettyxml(indent="  ")
```